ui-v2/app/services/repository/token.js

```diff
@@ -20,10 +21,22 @@ export default Service.extend({
     return status(obj);
   },
   self: function(secret, dc) {
-    return get(this, 'store').self(this.getModelName(), {
-      secret: secret,
-      dc: dc,
-    });
+    return get(this, 'store')
+      .self(this.getModelName(), {
+        secret: secret,
+        dc: dc,
+      })
+      .catch(e => {
+        // If we get this 500 RPC error, it means we are a legacy ACL cluster
+        // set AccessorID to null - which for the frontend means legacy mode
+        if (e.errors[0].detail.indexOf(UNKNOWN_METHOD_ERROR) === 0) {
```

Comment:
I guess this works. It would be nice if the error from the server was a little more explicit.

Reply:
Hmm yeah it's not great, the entire message is actually:

`rpc error making call: rpc: can't find method ACL.TokenList`

or

`rpc error making call: rpc: can't find method ACL.TokenRead`

I'm gonna change what I'm searching for to:

`rpc error making call: rpc: can't find method ACL`

I think, so add `ACL` just incase

Reply:
i think you want `!== -1` so that middle-of-the-string matches work.  When proxying between datacenters you will accumulate extra `rpc error making call:` prefixes for each hop

Reply:
So what will that error look like?

`rpc error making call: rpc error making call: rpc error making call: rpc: can't find method ACL.TokenRead`

?